compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/FirDataFlowAnalyzer.kt

```diff
@@ -155,13 +151,10 @@ abstract class FirDataFlowAnalyzer<FLOW : Flow>(
         return variable.stability to types.toMutableList()
     }
 
-    fun returnExpressionsOfAnonymousFunction(function: FirAnonymousFunction): Collection<FirStatement> {
+    fun returnExpressionsOfAnonymousFunction(function: FirAnonymousFunction): Collection<FirStatement>? {
```

Comment:
This is actually a hint that the usages are somewhat strange - there's currently some code duplication and inconsistency in how return types of lambdas are inferred. @sfs was looking at this in the context of `@Composable` functions, I think. I tried to modify that code too, but ended up breaking several hacks in checkers so that's nowhere close to being ready.

Reply:
FYI, here's one example of how these inconsistencies manifest: https://youtrack.jetbrains.com/issue/KT-54742. Note how there is apparently code to handle this exact case, it just doesn't work for...reasons.

Reply:
I didn't get how it's connected to the `returnExpressionsOfAnonymousFunction(orNull)`
Most users of `returnExpressionsOfAnonymousFunction` assumes that lambda already analyzed, so DFA knows all return expressions. And there is only one exception when this function can be called before/during lambda analysis. So I don't understand why exceptional behavior should be declared as default one in API)